lib/bot/bot.js

```diff
@@ -182,6 +183,45 @@ externals.Bot = class {
       parsedMessage.message.commandPrefix || _.toUpper(this.getId()) ===
       parsedMessage.message.commandPrefix;
     const isSelf = parsedMessage.user === _.toUpper(this.getId());
+    const options = {
+      botToken: this.config.botToken,
+      agent: this.getHttpAgent(),
+      user: parsedMessage.user,
+    };
+    let isAllowedUser = false;
+    let slackUser;
+
+    try {
+      slackUser = await this.fetchUserInfo(options);
+    } catch(err) {
+      logger.error(err);
+    }
+
+    if (this.config.allowedADGroupsOnly) {
+      const ad = new ActiveDirectory(this.config.activeDirectory);
+
+      _.forEach(this.config.allowedADGroupsOnly, (groupName)=> {
+        ad.isUserMemberOf(slackUser.user.name, groupName, (err, isMember)=> {
```

Comment:
`slackUser.user.name` is the name the user sets by himself or can be changed. I think email is a better attribute to use. Can you consider using `ad.findUser` method and assert if the user are part of any of the group from `this.config.allowedADGroupsOnly`?